memfault/sdk-nrf
# fix(platform-config): Rename platform config as extra config

fix(platform-config): Rename platform config as extra config

 ### Summary

A new Kconfig option was added to Memfault's Zephyr port,
`CONFIG_MEMFAULT_PLATFORM_EXTRA_CONFIG_FILE=y`, to address
losing a user's platform configurations set in `memfault_platform_config.h`, which
[happened in the nRF Connect SDK recently](https://devzone.nordicsemi.com/f/nordic-q-a/103188/memfault-sdk-integration-missing-user-memfault_platform_config). This commit makes
use of that Kconfig option by renaming the platform config file and including
it in the build if that option is set.

 ### Test Plan

Manually tested `CMakeLists.txt` change:
- Build fails as expected if `CONFIG_MEMFAULT_PLATFORM_EXTRA_CONFIG_FILE=y`,
`CONFIG_MEMFAULT_METRICS_EXTRA_DEFS_FILE=n` (forced this  by
commenting out a `select` line) and NCS platform file is NOT renamed. This
makes sure that the `zephyr_include_directories_ifdef()` for 
`CONFIG_MEMFAULT_METRICS_EXTRA_DEFS_FILE` including the `config/` folder
isn't causing a false positive
- Build succeeds and user `memfault_platform_config.h` settings are included
if same settings as above but NCS file is renamed to `memfault_platform_config_extra.h`.
Successful include of user platform config was checked by setting the heartbeat interval to 1,
and checking for the event overflow within a few seconds of boot
